libs/sync/sync_test.go

```diff
@@ -165,3 +168,41 @@ func TestRecursiveExclude(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, len(fileList), 7)
 }
+
+func TestNegateExclude(t *testing.T) {
+	ctx := context.Background()
+
+	dir := setupFiles(t)
+	root := vfs.MustNew(dir)
+	fileSet, err := git.NewFileSet(root)
+	require.NoError(t, err)
+
+	err = fileSet.EnsureValidGitIgnoreExists()
+	require.NoError(t, err)
+
+	inc, err := fileset.NewGlobSet(root, []string{})
+	require.NoError(t, err)
+
+	excl, err := fileset.NewGlobSet(root, []string{"./*", "!*.txt"})
+	require.NoError(t, err)
+
+	s := &Sync{
+		SyncOptions: &SyncOptions{},
+
+		fileSet:        fileSet,
+		includeFileSet: inc,
+		excludeFileSet: excl,
+	}
+
+	fileList, err := s.GetFileList(ctx)
+	require.NoError(t, err)
+	require.Equal(t, len(fileList), 1)
+	found := false
+	for _, f := range fileList {
```

Comment:
We don't need the for loop here. We can assert on the content directly since we already know it's of length 1.

Reply:
Indeed, good catch